components/wifi/esp8266-driver/ESP8266/ESP8266.cpp

```diff
@@ -964,20 +1020,33 @@ bool ESP8266::close(int id)
                     _closed = false;
                     _sock_i[id].open = false;
                     _clear_socket_packets(id);
+                    // Closed, so this socket escapes from SEND FAIL status.
+                    if (id == _sock_sending_id) {
+                        _sock_sending_id = -1;
+                    }
+                    _sock_i[id].send_fail = false;
                     _smutex.unlock();
                     // ESP8266 has a habit that it might close a socket on its own.
+                    tr_debug("close(%d): socket close OK with UNLINK ERROR", id);
                     return true;
                 }
             } else {
                 // _sock_i[id].open set to false with an OOB
                 _clear_socket_packets(id);
+                // Closed, so this socket escapes from SEND FAIL status
+                if (id == _sock_sending_id) {
+                    _sock_sending_id = -1;
+                }
+                _sock_i[id].send_fail = false;
```

Comment:
This same snippet has been added at least to 6 different places so maybe it should be turned into a function.

Reply:
Added a function and replaced the reoccurent code.